denlillemand/week3/ass4/app/namedlist.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import Message from './message.jsx';
+
+export default class NamedList extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.state = {
+            messageSwapDestination: '',
+            message: '',
+            sortedMessages: [],
+            messageSwapId: '',
+            messageId: 1,
+        };
+        this.onMessageSubmit = this.onMessageSubmit.bind(this);
+        this.onChangeMessageSwapDestinationListName = this.onChangeMessageSwapDestinationListName.bind(this);
+        this.onChangeMessage = this.onChangeMessage.bind(this);
+        this.onChangeMessageSwapId = this.onChangeMessageSwapId.bind(this);
+        this.archiveMessage = this.archiveMessage.bind(this);
+        this.deleteMessage = this.deleteMessage.bind(this);
+        this.swapMessage = this.swapMessage.bind(this);
+    }
+
+    swapMessage(event) {
+        this.props.swapMessage(parseInt(this.state.messageSwapId), this.state.messageSwapDestination, this.props.name);
+        this.setState({
+            messageSwapId: '',
+            messageSwapDestination: '',
+        });
+    }
+
+    onChangeMessage(event) {
+        if (event.target.value.length > 200) {
+            alert('Message can atmost be 200 characters long');
+        } else {
+            this.setState({
+                message: event.target.value,
+            });
+        }
+    }
+
+    onChangeMessageSwapDestinationListName(event) {
+        this.setState({
+            messageSwapDestination: event.target.value,
+        });
+    }
+
+    onMessageSubmit(event) {
+        this.props.onSubmitMessage({ text: this.state.message, id:this.state.messageId, archived: false }, this.props.name);
+        this.setState({
+            message: '',
+            messageId: this.state.messageId + 1,
+        });
+    }
+
+    onChangeMessageSwapId(event) {
+        if (parseInt(event.target.value)) {
```

Comment:
It is recommended that you specify a radix to the `parseInt` function. If a radix is not specified it usually defaults to 10, but on different javascript engines, this can lead to different behavior. See doc https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt